examples/software-device/rs-software-device.cpp

```diff
@@ -0,0 +1,185 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include <librealsense2/rs.hpp> // Include RealSense Cross Platform API
+#include <librealsense2/hpp/rs_internal.hpp>
+
+#include "example.hpp" 
+#include "example.hpp"          // Include short list of convenience functions for rendering
+
+
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <../third-party/stb_image_write.h>
+#include <../common/res/int-rs-splash.hpp>
+
+#define STB_IMAGE_IMPLEMENTATION
+#include <../third-party/stb_image.h>
+
+using namespace rs2;
+
+struct synthetic_frame
+{
+    int x, y, bpp;
+    std::vector<uint8_t> frame;
+};
+
+synthetic_frame create_synthetic_texture()
+{
+    synthetic_frame frame;
+    auto r = stbi_load_from_memory(splash, (int)splash_size, &frame.x, &frame.y, &frame.bpp, false);
+
+    std::vector<uint8_t> pixels_color(frame.x * frame.y * frame.bpp, 0);
+    
+    memcpy(pixels_color.data(), r, frame.x*frame.y * 4);
+
+    for (auto i = 0; i< frame.y; i++)
+        for (auto j = 0; j < frame.x * 4; j += 4)
+        {
+            if (pixels_color.data()[i*frame.x * 4 + j] == 0)
+            {
+                pixels_color.data()[i*frame.x * 4 + j] = 22;
+                pixels_color.data()[i*frame.x * 4 + j + 1] = 115;
+                pixels_color.data()[i*frame.x * 4 + j + 2] = 185;
+            }
+        }
+    frame.frame = pixels_color;
+
+    return frame;
+}
+
+void fill_synthetic_depth_data(void* data, int w, int h, int bpp, float wave_base)
+{
+    for (int i = 0; i < h; i++)
+    {
+        for (int j = 0; j < w; j++)
+        {
+            auto d = 2 + 0.2 * 0.5 * (1 + sin(wave_base + j / 50.f));
+            ((uint16_t*)data)[i*w + j] = (int)(d * 0xff);
+        }
+    }
+}
+
+int main(int argc, char * argv[]) try
+{
+    const int W = 640;
+    const int H = 480;
+    const int BPP_D = 2;
+
+    rs2_intrinsics depth_intrinsics{ W, H, W / 2, H / 2, W , H , RS2_DISTORTION_BROWN_CONRADY ,{ 0,0,0,0,0 } };
+
+    auto texture = create_synthetic_texture();
+
+    rs2_intrinsics color_intrinsics = { texture.x, texture.y, 
+        (float)texture.x / 2, (float)texture.y / 2,
+        (float)texture.x / 2, (float)texture.y / 2,
+        RS2_DISTORTION_BROWN_CONRADY ,{ 0,0,0,0,0 } };
+
+   
+    //==================================================//
+    //           Declare Software-Only Device           //
+    //==================================================//
+    
+    software_device dev; // Create software-only device
```

Comment:
I believe most users of the `software_device` will want to have a `MyDevice` which is a `software_device`, have you considered making the example show this?

Reply:
This will also encapsulate the  `//           Declare Software-Only Device           //` section into an object